pennylane/transforms/broadcast_expand.py

```diff
@@ -112,7 +112,12 @@ def broadcast_expand(tape):
 
     def processing_fn(results):
         if len(tape.measurements) > 1 and qml.active_return():
-            results = [qml.math.hstack(r) for r in results]
+            processed_results = [None] * len(tape.measurements)
```

Comment:
why are you pre-allocating `None` values instead of just using a list-comprehension below?

Reply:
Good point, updated!